utils.py

```diff
@@ -87,11 +87,25 @@ def handle_user_choice(data, canvas_bounding_boxes):
         ocr_bounding_box = data["objects"][i]
         if canvas_bounding_box["fill"] == "rgb(208, 240, 192, 0.2)":
             ocr_bounding_box["result"] = True
+            ocr_bounding_box["first"] = True
+            ocr_bounding_box["last"] = True
+        elif canvas_bounding_box["fill"] == "rgb(208, 237, 192, 0.2)":
+            ocr_bounding_box["first"] = True
+            ocr_bounding_box["result"] = True
+        elif canvas_bounding_box["fill"] == "rgb(208, 238, 192, 0.2)":
+            ocr_bounding_box["result"] = True
+        elif canvas_bounding_box["fill"] == "rgb(208, 239, 192, 0.2)":
```

Comment:
Are you checking 4 times the same condition ?

Reply:
No we are trying to get back whether it's the first or the last box (if found using the selection tool). We do it using the color of those boxes (which is why there are slightly different)

We are trying to see if we can get back a group number or something like that